skyportal/handlers/api/classification.py

```diff
@@ -0,0 +1,239 @@
+                      classifications ought to equal unity. Only individual
+                      probabilities are checked.
+                  group_ids:
+                    type: array
+                    items:
+                      type: integer
+                    description: |
+                      List of group IDs corresponding to which groups should be
+                      able to view classification. Defaults to all of
+                      requesting user's groups.
+                required:
+                  - obj_id
+                  - classification
+                  - taxonomy_id
+        responses:
+          200:
+            content:
+              application/json:
+                schema:
+                  allOf:
+                    - $ref: '#/components/schemas/Success'
+                    - type: object
+                      properties:
+                        data:
+                          type: object
+                          properties:
+                            comment_id:
+                              type: integer
+                              description: New classification ID
+        """
+        data = self.get_json()
+        obj_id = data['obj_id']
+        # Ensure user/token has access to parent source
+        _ = Source.get_obj_if_owned_by(obj_id, self.current_user)
+        user_group_ids = [g.id for g in self.current_user.groups]
+        group_ids = data.pop("group_ids", user_group_ids)
+        group_ids = [gid for gid in group_ids if gid in user_group_ids]
+        if not group_ids:
+            return self.error(f"Invalid group IDs field ({group_ids}): "
+                              "You must provide one or more valid group IDs.")
+        groups = Group.query.filter(Group.id.in_(group_ids)).all()
+
+        author = self.associated_user_object.username
+
+        # check the taxonomy
+        taxonomy_id = data["taxonomy_id"]
+        taxonomy = Taxonomy. \
+            get_taxonomy_usable_by_user(taxonomy_id, self.current_user)
+        if taxonomy is None:
+            return self.error(
+                'That taxonomy does not exist or is not available to user.'
+            )
+        if not isinstance(taxonomy, list):
+            return self.error('Problem retreiving taxonomy')
+
+        if data['classification'] not in taxonomy[0].allowed_classes:
+            return self.error(f"That classification ({data['classification']}) "
+                               'is not in the allowed classes for the chosen '
+                               f'taxonomy (id={taxonomy_id}')
+
+        probability = data.get('probability')
+        if probability is not None:
+          if probability < 0 or probability > 1:
+            return self.error(f"That probability ({probability}) is outside "
+                               "the allowable range (0-1).")
+
+        classification = Classification(classification=data['classification'],
```

Comment:
Concretely, what does it mean for a classification to be associated with a group?

Reply:
It means that users in that group can see the classification, otherwise they cannot.

Reply:
Classifications are competition sensitive, even more than for comments.